### ISSUE TITLE
Expo updates causing crash on load in SDK 39

### ISSUE THREAD

**Issue by TMcLoone** [expo/expo#10286]
### Summary of Issue 
Updating to expo39 and the latest expo-updates causes a crash on loading with the following error:

NSInvalidArgumentException
-[EXUpdatesAppController appLoaderTask:didFinishBackgroundUpdateWithStatus:update:error:]: unrecognized selector sent to instance
 File "node_modules/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesAppLoaderTask.m", line 291, in __78-[EXUpdatesAppLoaderTask _didFinishBackgroundUpdateWithStatus:manifest:error:]_block_invoke

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
  Expo CLI 3.27.7 environment info:
    System:
      OS: macOS 10.15.6
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 14.3.0 - ~/.nvm/versions/node/v14.3.0/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.8 - ~/.nvm/versions/node/v14.3.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.9.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.0, DriverKit 19.0, macOS 10.15, tvOS 14.0, watchOS 7.0
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6626763
      Xcode: 12.0/12A7209 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^39.0.0 => 39.0.2
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: 0.63.2 => 0.63.2
    npmGlobalPackages:
      expo-cli: 3.27.7
    Expo Workflow: bare

Targeting iOS
 
### Steps to Reproduce

Updated to expo sdk 39 and latest expo-updates. Build and run on device (release)
### Expected Behavior vs Actual Behavior
Expected app to load but crashes after splash screen

**Comment by tonnenpinguin** [comment]
We're running into the same issue

**Comment by sjchmiela** [comment]
Hey, could you try replacing

```diff
-- (void)appLoaderTask:(EXUpdatesAppLoaderTask *)appLoaderTask didCompleteBackgroundUpdateWithStatus:(EXUpdatesBackgroundUpdateStatus)status update:(nullable EXUpdatesUpdate *)update error:(nullable NSError *)error
+- (void)appLoaderTask:(EXUpdatesAppLoaderTask *)appLoaderTask didFinishBackgroundUpdateWithStatus:(EXUpdatesBackgroundUpdateStatus)status update:(nullable EXUpdatesUpdate *)update error:(nullable NSError *)error
```

in `EXUpdatesAppController.m`? (Basically `didCompleteBackgroundUpdateWithStatus` for `didFinishBackgroundUpdateWithStatus`.)

https://github.com/expo/expo/blob/1cecb7d2df9a25ed3e7afd008e79352a9f66a8fa/packages/expo-updates/ios/EXUpdates/EXUpdatesAppController.m#L231

**Comment by TMcLoone** [comment]
@sjchmiela  That fixed it 🎉

**Comment by esamelson** [comment]
Thanks all for the report, and to @sjchmiela for the fix! Released in 0.3.4.
